interface.go

```diff
@@ -94,51 +96,48 @@ func (c *Cache) GetOrCompute(key string, evaluator func() (interface{}, error))
 	return value, err
 }
 
-// DeleteWithPrefix removes all keys that start with given prefix
-func (c *Cache) DeleteWithPrefix(prefix string) error {
+//DeletePredicate deletes all keys matching the supplied predicate, returns number of deleted keys
+func (c *Cache) DeletePredicate(pred Predicate) (int, error) {
+	count := 0
+
 	keys, err := c.Keys()
 	if err != nil {
-		return err
+		return count, err
 	}
+
 	for _, key := range keys {
-		if strings.HasPrefix(key, prefix) {
+		if pred(key) {
 			if err := c.Delete(key); err != nil {
-				return err
+				return count, err
 			}
 		}
+		count++
```

Comment:
I think this is incremented in the wrong closure. You count all the keys and I think that you should only count the keys that match the predicate